### PR TITLE
refactor: code refactor, base unit test setup, sample unit test for user service

### LINKED ISSUE #59: [FEAT] Setup and Configure Unit Testing
## Description
Implement setup for unit testing. This is to aid developers in implementing, testing, and analyzing their codes, as well as illustrate that their implementation and requirements are meant. This will also help to unblock devs waiting on db before implementing their features

## Acceptance Criteria

- [ ] Unit test project setup

## Purpose
Aid team members in implementing, testing, and illustrating their codes

## Requirements

- Setup and configure unit test projects using the NUnit framework

### PR DESCRIPTION
**Pull request by dmendie**
# Description

Implement setup for unit testing. This is to aid developers in implementing, testing, and analyzing their codes, as well as illustrate that their implementation and requirements are meant. This will also help to unblock devs waiting on db before implementing their features

**Closes #issue_number_here**
Closes #59

# Changes proposed
Added base unit test class

## What were you told to do?

Implement and set up unit test features for services

## What did you do?

1. code refactor - moved service dependencies to one class file for easy referencing in the test class library
2. base unit test setup - set base test class that has all the necessary configurations to test service implementation with mock data
3. added sample unit test for user service - added a sample UserServicetest class to illustrate unit test cases for the user service

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

**Test Results**
Capture

**Video Link**
https://drive.google.com/file/d/1RHOBw27XoIv8GVUdcu29mvqdDRuJoquy/view?usp=sharing

### CODE REVIEW

File: src/Hng.Web/Bootstrappers/ServicesBootstrapper.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using Hng.Application.Services;
+using Hng.Domain.Models;
+using Hng.Infrastructure.Repository.Interface;
+using Hng.Infrastructure.Repository.Mock;
+using Hng.Infrastructure.Repository.SqlData;
+using Hng.Web.Mappers;
+
+namespace Hng.Web.Bootstrappers
+{
+    public static class ServicesBootstrapper
```
Comment (NdubuisiJr) [inline src/Hng.Web/Bootstrappers/ServicesBootstrapper.cs:11]: What in christ's name is this?

File: src/Hng.Application.Test/Hng.Application.Test.csproj [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
-    <Nullable>enable</Nullable>
-  </PropertyGroup>
+	<PropertyGroup>
+		<TargetFramework>net8.0</TargetFramework>
+		<ImplicitUsings>enable</ImplicitUsings>
+		<Nullable>enable</Nullable>
+	</PropertyGroup>
 
-  <ItemGroup>
-    <PackageReference Include="coverlet.collector" Version="6.0.2">
```
Comment (NdubuisiJr) [inline src/Hng.Application.Test/Hng.Application.Test.csproj:10]: Why did you remove `Xunit`?

**Review by NdubuisiJr — CHANGES_REQUESTED** [review]

**Comment by NdubuisiJr** [comment]
Who approved this ticket? This doesn't even add any value to the system

File: src/Hng.Web/Bootstrappers/ServicesBootstrapper.cs (hunk 1 above)
Comment (dmendie) [reply src/Hng.Web/Bootstrappers/ServicesBootstrapper.cs:11]: the term "bootstrapper" refers to a process or a set of initial steps necessary to start a larger and more complex process. Registration of repositories and services is one of those initial steps. The reason this file exists is to prevent having to register repositories and core services in different locations since the web needs it and also the unit test application

Team members only need to add their services to this file once and it is available in any part of the application where registration is required (test and web project)

This file is just a dedicated place for registering repositories and custom services. The naming might not be right and you can suggest a name and I'll change it to that

File: src/Hng.Application.Test/Hng.Application.Test.csproj (hunk 2 above)
Comment (dmendie) [reply src/Hng.Application.Test/Hng.Application.Test.csproj:10]: I needed a framework suitable for projects that rely heavily on data-driven tests. To be able to use [SetUp], [TearDown], [OneTimeSetUp], and [OneTimeTearDown] attributes for setup and teardown logic at the test and fixture levels rather than using constructors or IDisposable.

**Comment by dmendie** [comment]
Every ticket requires a unit test coverage as part of its deliverable and what this ticket addressed was setting up and configuring the base test structure for other tests to be written in for the boiler_plate_csharp project.

One value this adds to the system is that it helps to unblock developers by allowing them to implement their service logics using mock while waiting for their database schema to be created by db moderators
